packages/account/src/Sections/Profile/PersonalDetails/__tests__/personal-details.spec.tsx

```diff
@@ -0,0 +1,222 @@
+import { PersonalDetailsForm } from '../personal-details';
+import { StoreProvider, mockStore } from '@deriv/stores';
+import userEvent from '@testing-library/user-event';
+
+afterAll(cleanup);
+jest.mock('@deriv/components', () => ({
+    ...jest.requireActual('@deriv/components'),
+    Loading: () => <div>Loading</div>,
+}));
+
+jest.mock('@deriv/shared/src/services/ws-methods', () => ({
+    __esModule: true, // this property makes it work,
+    default: 'mockedDefaultExport',
+    WS: {
+        wait: (...payload: []) => {
+            return Promise.resolve([...payload]);
+        },
+    },
+    useWS: () => undefined,
+}));
+
+describe('<PersonalDetailsForm />', () => {
+    const history = createBrowserHistory();
+
+    const promise = Promise.resolve();
+    const fetchResidenceList = jest.fn(() => promise);
+    const fetchStatesList = jest.fn(() => promise);
+    const residence_list = [
+        {
+            text: 'Text',
+            value: 'value',
+        },
+    ];
+    const mock_store = mockStore({
+        client: {
+            account_settings: {
+                first_name: 'John',
+                place_of_birth: 'Thailand',
+                citizen: 'Thailand',
+                email_consent: 1,
+            },
+            is_virtual: false,
+            states_list: residence_list,
+            residence_list,
+            has_residence: true,
+            getChangeableFields: () => [],
+            fetchResidenceList,
+            fetchStatesList,
+        },
+    });
+
+    const renderComponent = (modified_store = mock_store) => {
+        return render(
+            <Router history={history}>
+                <StoreProvider store={modified_store}>
+                    <PersonalDetailsForm history={history} />
+                </StoreProvider>
+            </Router>
+        );
+    };
+
+    it('should render successfully', async () => {
+        renderComponent();
+        expect(screen.queryByText(/Loading/)).not.toBeInTheDocument();
+        await screen.findByText(
+            /Please make sure your information is correct or it may affect your trading experience./i
+        );
+    });
+
+    it('should render all the personal details fields', () => {
+        renderComponent();
+        const fields = [
+            'First name*',
+            'Last name*',
+            // 'Place of birth',
+            'Date of birth*',
+            // 'Citizenship',
+            'Country of residence*',
+            'Phone number*',
+            'First line of address*',
+            'Second line of address (optional)',
+            'Town/City*',
+            'State/Province (optional)',
+            'Postal/ZIP code',
+        ];
+        fields.forEach(value => {
+            expect(screen.getByText(value)).toBeInTheDocument();
+        });
+    });
+
+    it('should have "required" validation errors on required form fields', async () => {
+        renderComponent();
+
+        await waitFor(() => {
```

Comment:
do we really need waitFor() here and in tests below?
can we just findByText asynchronously? :)

Reply:
Tried to use findByText but the test case fails